ci/kind/test-e2e-kind.sh

```diff
@@ -18,31 +18,79 @@
 
 set -eo pipefail
 
+function echoerr {
+    >&2 echo "$@"
+}
+
+_usage="Usage: $0 [--encap-mode <mode>] [--proxy] [--np] [--help|-h]
+        --encap-mode                  Traffic encapsulation mode. (default is 'encap')
+        --proxy                       Enables Antrea proxy.
+        --np                          Enables Namespaced Antrea NetworkPolicy CRDs and ClusterNetworkPolicy related CRDs.
+        --help, -h                    Print this message and exit
+"
+
+function print_usage {
+    echoerr "$_usage"
+}
+
+
 TESTBED_CMD=$(dirname $0)"/kind-setup.sh"
 YML_CMD=$(dirname $0)"/../../hack/generate-manifest.sh"
 COMMON_IMAGES="busybox nginx antrea/antrea-ubuntu:latest"
 
 function quit {
   if [[ $? != 0 ]]; then
-    echo " Test failed cleaning testbed"
+    echoerr " Test failed cleaning testbed"
     $TESTBED_CMD destroy kind
   fi
 }
 trap "quit" INT EXIT
 
-function run_test {
-  mode=$1
-  proxy=$2
-  args=$3
-  if [[ $proxy != "--proxy" ]]; then
-    proxy=""
-    args=$2
-  fi
+mode=""
+proxy=false
+np=false
+args=""
+while [[ $# -gt 0 ]]
+do
+key="$1"
+
+case $key in
+    --proxy)
+    proxy=true
+    shift
+    ;;
+    --np)
+    np=true
+    shift
+    ;;
+    --encap-mode)
+    mode="$2"
+    shift 2
+    ;;
+    -h|--help)
+    print_usage
+    exit 0
+    ;;
+    *)    # unknown option
+    args="$args $1"
+    ;;
+esac
+done
+
+manifest_args=""
+if $proxy; then
+    # See https://github.com/vmware-tanzu/antrea/issues/897
+    manifest_args="$manifest_args --proxy --tun vxlan"
```

Comment:
These should be for np, not proxy?

Reply:
correct, fixed

Reply:
i think you need to switch proxy with np